caddy/setup/locale.go

```diff
@@ -0,0 +1,79 @@
+package setup
+
+import (
+	"strings"
+
+	"github.com/mholt/caddy/middleware"
+	"github.com/mholt/caddy/middleware/locale"
+	"github.com/mholt/caddy/middleware/locale/method"
+)
+
+// Locale configures a new Locale middleware instance.
+func Locale(c *Controller) (middleware.Middleware, error) {
+	locale := &locale.Locale{}
+
+	locales, methods, settings, err := localeParse(c)
+	if err != nil {
+		return nil, err
+	}
+
+	return func(next middleware.Handler) middleware.Handler {
+		locale.Next = next
+		locale.Locales = locales
+		locale.Methods = methods
+		locale.Settings = settings
+		return locale
+	}, nil
+}
+
+func localeParse(c *Controller) ([]string, []method.Method, *method.Settings, error) {
+	locales := []string{}
+	methods := []method.Method{}
+	settings := &method.Settings{}
+
+	for c.Next() {
+		args := c.RemainingArgs()
+
+		switch len(args) {
+		default:
+			locales = append(locales, args...)
+			fallthrough
```

Comment:
True! I copied that from `caddy/setup/rewrite.go`. Maybe that can be refactored too.